### PR TITLE
Adding multi GPU support and several other enhancements

### PR DESCRIPTION
**Pull request by jammy270**
**Major changes**

1. Adding multi GPU support during training
2. Parallelized the code for deepmrseg_test which reduces the inference time drastically for MUSE
3. Inference during validation/testing now accepts larger batch sizes (specfically, batch_size*8)
4. Modified the loss function to a combo loss with a loss weight parameter (alpha)
5. Added tf.summary outputs that can be visualized via tensorboard
6. Split predictClasses function to better interface with DeepMRSegApp

Resolves issues #20 #21 #22 #23 #24 #25

### CODE REVIEW

**Review by ashishsingh18 — APPROVED** [review]
Looks fine. Codacy issues will be handled separately.

**Comment by jammy270** [comment]
Shouldn't the issues 20-25 be resolved with this pull request? They still show as open...

**Comment by ashishsingh18** [comment]
> 
> 
> Shouldn't the issues 20-25 be resolved with this pull request? They still show as open...

Issues need to be linked to the PR as per the guidelines here: https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue 

We can manually close them for now.

**Comment by jammy270** [comment]
> > Shouldn't the issues 20-25 be resolved with this pull request? They still show as open...
> 
> Issues need to be linked to the PR as per the guidelines here: https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue
> 
> We can manually close them for now.

Hmm. I thought I did link the issues to this PR and we can see they were linked successfully. Maybe because 1 codacy check failed, the issues remained open...

**Comment by ashishsingh18** [comment]
> 
> 
> > > Shouldn't the issues 20-25 be resolved with this pull request? They still show as open...
> > 
> > 
> > Issues need to be linked to the PR as per the guidelines here: https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue
> > We can manually close them for now.
> 
> Hmm. I thought I did link the issues to this PR and we can see they were linked successfully. Maybe because 1 codacy check failed, the issues remained open...

Not sure if this has to do with Codacy. I usually use the full syntax when linking multiple issues and that has always worked for me.
